libmeepgeom/cyl-ellipsoid-ll.cpp

```diff
@@ -0,0 +1,121 @@
+/*             P for TM polarization                           */
+/***************************************************************/
+int main(int argc, char *argv[])
+{
+  initialize mpi(argc, argv);
+
+  // simple argument parsing 
+  meep::component src_cmpt=Ez;
+  for(int narg=1; narg<argc; narg++)
+   if ( argv[narg] && !strcmp(argv[narg],"--polarization") )
+    { if (narg+1 == argc) 
+       meep::abort("no option specified for --polarization");
+      else if (!strcasecmp(argv[narg+1],"S"))
+       printf("Using S-polarization\n");
+      else if (!strcasecmp(argv[narg+1],"P"))
+       { src_cmpt=Hz;
+         printf("Using P-polarization\n");
+       }
+      else 
+       meep::abort("invalid --polarization %s",argv[narg+1]);
+    };
+
+  //(set-param! resolution 100)
+  double resolution = 100.0; 
+
+  // (set! geometry-lattice (make lattice (size 10 10 no-size)))
+  // (set! pml-layers (list (make pml (thickness 1))))
+  // (if (= src-cmpt Ez)
+  //  (set! symmetries (list (make mirror-sym (direction X))
+  //                         (make mirror-sym (direction Y)))))
+  // (if (= src-cmpt Hz) 
+  //  (set! symmetries (list (make mirror-sym (direction X) (phase -1))
+  //  (set! symmetries (list (make mirror-sym (direction Y) (phase -1))
+  geometry_lattice.size.x=10.0;
+  geometry_lattice.size.y=10.0;
+  geometry_lattice.size.z=0.0;
+  grid_volume gv = voltwo(10.0, 10.0, resolution);
+  gv.center_origin();
+ // symmetry sym = (src_cmpt==Ez) ?  mirror(X,gv) + mirror(Y,gv)
+ //                               : -mirror(X,gv) - mirror(Y,gv);
+  symmetry sym = identity();
+  structure the_structure(gv, dummy_eps, pml(1.0), sym);
+
+  // (set! geometry (list 
+  //    (make cylinder (center 0 0 0) (radius 3) (height infinity)
+  //                   (material (make medium (index 3.5))))
+  //    (make ellipsoid (center 0 0 0) (size 1 2 infinity)
+  //                   (material air))))
+  double n=3.5; // index of refraction
+  material_type dielectric = meep_geom::make_dielectric(n*n);
+  geometric_object objects[2];
+  vector3 center = {0.0, 0.0, 0.0};
+  double radius  = 3.0;
+  double height  = 1.0e20;
+  vector3 xhat   = {1.0, 0.0, 0.0};
+  vector3 yhat   = {0.0, 1.0, 0.0};
+  vector3 zhat   = {0.0, 0.0, 1.0};
+  vector3 size   = {1.0, 2.0, 1.0e20};
+  objects[0] = make_cylinder(dielectric, center, radius, height, zhat);
+  objects[1] = make_ellipsoid(meep_geom::vacuum, center, xhat, yhat, zhat, size);
+  geometric_object_list g={ 2, objects };
+  meep_geom::set_materials_from_geometry(&the_structure, g);
+
+  // (set! sources (list (make source (src (make gaussian-src (frequency 1) (fwidth 0.1)))
+                     //  (center 0 0 0) (component src-cmpt))))
+  fields f(&the_structure);
+  double fcen = 1.0;
+  double df   = 0.1;
+  gaussian_src_time src(fcen, df);
+  vec src_point = vec(0.0, 0.0);
+  vec src_size  = vec(10.0, 10.0);
+  f.add_volume_source(src_cmpt, src, volume(src_point, src_size));
```

Comment:
Thanks! That solves the problem---here's what my field output looks like now:

![ez-000004600](https://user-images.githubusercontent.com/920123/27999380-83dda0f2-64ea-11e7-802c-6fd611776312.png)